### PR TITLE
universal cache dir

### PR DESCRIPTION
**Pull request by jxmorris12**
- get rid of `config.yaml`
- tensorflow hub, transformers, etc. use TA_CACHE_DIR
- add `pip install "[.dev]"` option

### CODE REVIEW

File: textattack/commands/attack/attack_command.py [hunk 1]
```diff
@@ -157,6 +157,24 @@ def str_to_int(s):
             default=float("inf"),
             help="The maximum number of model queries allowed per example attacked.",
         )
+        parser.add_argument(
```
Comment (jinyongyoo) [inline textattack/commands/attack/attack_command.py:160]: When would the user ever need to set the batch size when attacking? We don't process attacks in batches.

File: textattack/shared/utils/install.py [hunk 2]
```diff
@@ -122,28 +120,39 @@ def _post_install():
     nltk.download("stopwords")
 
 
+def set_cache_dir(cache_dir):
+    """ Sets all relevant cache directories to ``TA_CACHE_DIR``. """
+    # Tensorflow Hub cache directory
+    os.environ["TFHUB_CACHE_DIR"] = cache_dir
```
Comment (jinyongyoo) [inline textattack/shared/utils/install.py:126]: I'm not sure if this is necessarily good for the user. (1) They might not know that their cache location just got switched. (2) If they already have models and stuff cached, shouldn't we allow them to use it in the first place.

File: textattack/shared/utils/misc.py [hunk 3]
```diff
@@ -86,3 +86,4 @@ def set_seed(random_seed):
     random.seed(random_seed)
     np.random.seed(random_seed)
     torch.manual_seed(random_seed)
+    torch.cuda.manual_seed(random_seed)
```
Comment (jinyongyoo) [inline textattack/shared/utils/misc.py:89]: wow i didn't know we also had to set seed for CUDA. Good catch!

File: textattack/shared/attack.py [hunk 4]
```diff
@@ -145,18 +148,20 @@ def _filter_transformations(
         # Populate cache with transformed_texts
         uncached_texts = []
         for transformed_text in transformed_texts:
-            if transformed_text not in self.constraints_cache:
+            if (current_text, transformed_text) not in self.constraints_cache:
```
Comment (jinyongyoo) [inline textattack/shared/attack.py:151]: Very good catch.

File: textattack/shared/utils/misc.py (hunk 3 above)
Comment (jxmorris12) [reply textattack/shared/utils/misc.py:89]: ikr

File: textattack/shared/attack.py (hunk 4 above)
Comment (jxmorris12) [reply textattack/shared/attack.py:151]: :)

File: textattack/shared/utils/install.py (hunk 2 above)
Comment (jxmorris12) [reply textattack/shared/utils/install.py:126]: resolved w Jeffrey: In the future we'll add multiple levels/preferences for the cache directory

**Pull request merged**